From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: booky10 <boooky10@gmail.com>
Date: Fri, 5 Nov 2021 21:01:36 +0100
Subject: [PATCH] Add API for resetting a single score

It was only possible to reset all scores for a specific entry, instead of resetting only specific scores.

diff --git a/src/main/java/org/bukkit/scoreboard/Score.java b/src/main/java/org/bukkit/scoreboard/Score.java
index 8579c4f5e877a0db6df52094f2f2d82920f3c869..1eaa9a93f8eff5f18a6cce2d74f21eb19db273c8 100644
--- a/src/main/java/org/bukkit/scoreboard/Score.java
+++ b/src/main/java/org/bukkit/scoreboard/Score.java
@@ -73,4 +73,14 @@ public interface Score {
      */
     @Nullable
     Scoreboard getScoreboard();
+
+    // Paper start
+    /**
+     * Resets this score, if a value has been set.
+     *
+     * @throws IllegalStateException if the associated objective has been
+     *     unregistered
+     */
+    void resetScore() throws IllegalStateException;
+    // Paper end
 }
